### ISSUE TITLE
Is it a bug or a feature that validating starts before training ends?

### ISSUE THREAD

**Issue by lity** [PyTorchLightning/pytorch-lightning#1663]
pytorch-lightning version:0.7.5

1 gpu.

I found that validating started when training reached near end(about 98%), then training and validating run simultaneously.

Anyone ever noticed this?

I guess validating should start after training loop ends in every epoch. Am I wrong?

**Comment by lity** [comment]
It's my misunderstanding of progress bar of pytorch-lightning, sorry.

**Comment by AtomScott** [comment]
Hey, can you explain your misunderstanding?
I get confused with this sometimes too.

**Comment by AtomScott** [comment]
I think this should be somewhere in the docs, maybe in [lightningmodule/validation-loop](https://pytorch-lightning.readthedocs.io/en/stable/lightning_module.html#validation-loop).

Just witnessed colleague shy away from PL this wasn't clear.

**Comment by kaczmarj** [comment]
i found the answer here https://github.com/Lightning-AI/lightning/discussions/4133#discussioncomment-238367

>TL;DR - assuming you only look at the progress bar, it is the correct behavior, they don't run in parallel.
>
>In more details - AFAIK the progress bar shows the status of a full training + validation cycle, when both have the same size it does look like the validation starts in the middle of an epoch and that the validation set is half the size. Moreover, in the case of 2 gpus the number of steps for the full cycle will be the number of samples in your dataset however only half of the steps are required for full training epoch and the rest for validation.
